### PR TITLE
Added 2 string template functions

### PR DESCRIPTION
**Pull request by fredxinfan**
Added the following string template functions in "tpl/template.go":

1. Substr
Slice a string based on a specified range. e.g {{ substr "abc" 1 2 }} will return "b".

2. Split
Split a string based on a specified delimiter. e.g {{ split "a,b,c"  "," }} will return [a b c].

This is my first commit to Hugo. I've tried to follow the contributor guide. However, please feel free to let me know if I've missed anything. Cheers

### CODE REVIEW

**Comment by bep** [comment]
This is good stuff.

Two comments:

1) The `str` argument in the two functions should be `interface{}`. See `Trim`. With the templates you never know what type you get, it could be `[]byte`, `template.HTML` ...
2) Could you squash your commits into one or two logical units (with proper commit messages). `rebase -i master` or similar is a good starting point.

File: tpl/template_test.go [hunk 1]
```diff
@@ -276,6 +276,39 @@ func TestIn(t *testing.T) {
...
+	}
+}
+
+func TestSplit(t *testing.T) {
+	for i, this := range []struct {
+		v1 string
+		v2 string
+		expect []string
+	}{
+		{"a,b", ",", []string{"a", "b"}},
```
Comment (bep) [inline tpl/template_test.go]: You created a table-driven test, so you might as well add another line to this ... maybe another delim?

**Comment by bep** [comment]
This is a good resource for how to write good commit messages:

http://chris.beams.io/posts/git-commit/

(and yes, I know -- the commit log in Hugo doesn't live up to this standard, always ... but it is allowed to try ...)

**Comment by fredxinfan** [comment]
Hi @bep, thanks for all the advice. I've made some changes based on your feedback. Let me know if there are something more I need to improve.

Bookmarked the GIT article, very useful resource, appreciate it.

**Comment by bep** [comment]
Merged in 04817c7

I changed your commit message, the most important being the imperative "add" instead of "added" in the subject line.

**Comment by fredxinfan** [comment]
Ah ha, didn't get what dose the "imperative mood" mean when I read about it. Thanks for the change.

**Comment by anthonyfok** [comment]
Hi @fredxinfan, thank you for your contribution, and @bep, thank you for your commit!

After some thoughts, I think the current `substr` is a misnomer.  For most programmers coming from other languages like C++, PHP, JavaScript, etc., when they see `substr()`, they would think of *start_pos* and *length*, so `substr("abc", 1, 2)` should return `bc`, not just `b`.

What is currently implemented is not a `substr`, but rather, a slice.  The line `return aStr[pos:length], nil` in `func Substr` reflects that confusion.  It is not `aStr[pos:length]`, but rather `aStr[start:end]`.

I think we should fix this before v0.14, otherwise it would *really* confuse users when v0.14 is released.  (Well, it already confused me... it took me days to figure out why `{{ substr "abc" 1 2 }}` would return `b`, when all my previous training says it should return `bc`.

Suggested actions:

 * Rename what is currently implemented as `substr` to `slice` or something similar?

 * Implement a `substr` function that matches Hugo users' expectation?

Thanks!

**Comment by fredxinfan** [comment]
Hi @anthonyfok , I think what you suggested makes sense. The reason is good enough to me if someone already get confused by the function name, like you. The quickest way to resolve this is to rename ```substr``` to ```slice```. As for a more appropriate ```substr``` function, I'm not sure if it's necessary as they're really close enough to each other, but do let me know if you think it's important to have both. I will be happy to do it.

Cheers.

**Comment by bep** [comment]
@anthonyfok - agree, please rename.

**Comment by bep** [comment]
@fredxinfan see #991

**Comment by bep** [comment]
Reopening this ...

I think that with "Slice" people will also expect to slice bread and not just strings. :-)

We could *maybe* make it more generic, but I think Slicestr may be the correct name?

**Comment by anthonyfok** [comment]
@bep Good call about renaming `Slice` to `Slicestr`!

**Comment by fredxinfan** [comment]
Yep. Sounds good :-)

**Pull request closed without merging**